Cython/Utility/CythonFunction.c

```diff
@@ -1230,7 +1230,13 @@ static int __Pyx_CyFunction_InitClassCell(PyObject *cyfunctions, PyObject *class
 //@requires: CythonFunctionShared
 
 static int __Pyx_CyFunction_InitClassCell(PyObject *cyfunctions, PyObject *classobj) {
-    Py_ssize_t i, count = PyList_GET_SIZE(cyfunctions);
+    Py_ssize_t i, count;
+#if CYTHON_ASSUME_SAFE_MACROS
+    count = PyList_GET_SIZE(cyfunctions);
+#else
+    count = PyList_Size(cyfunctions);
+    if (count < 0) return -1;
+#endif
```

Comment:
I'm not sure that we're not overusing `CYTHON_ASSUME_SAFE_MACROS` in such cases. There's really no reason why `PyList_GET_SIZE()` would fail for a Python list object. Up for debate, but it *feels* excessive.

Reply:
Are you saying that this should probably just be doing `PyList_Size` for the limited API only, and there's no need to change this so it affects PyPy (for example)?

I'm happy to go with that - it's probably easier

Reply:
I've done all the comments but this one (which I'd like to check I've understood first)

Reply:
> Are you saying that this should probably just be doing `PyList_Size` for the limited API only, and there's no need to change this so it affects PyPy (for example)?

Yes. Let's stick with that pattern, until we have an actual reason to change it.

Reply:
To make it clear, this applies to the size reading, definitely. I'd keep the `SAFE_MACROS` guard for calls like `GET_ITEM()`, because PyPy, for example, really actually has to do something here in that it needs to allocate a `PyObject` representation of the Python object. So there are realistic reasons why this might fail. Reading a plain C value seems entirely harmless in comparison.

Reply:
I submitted https://github.com/cython/cython/pull/5882 to draw a clean line.